tools/setup-buildtools-apple.sh

```diff
@@ -9,10 +9,14 @@ echo ***
 echo *** You may need to enter your admin password to update the brew files:
 echo ***
 
-sudo chown -R $(whoami) /usr/local/Cellar
-sudo chown -R $(whoami) /usr/local/Homebrew
-sudo chown -R $(whoami) /usr/local/var/homebrew
-sudo chown -R $(whoami) /usr/local/etc/bash_completion.d /usr/local/include /usr/local/lib/pkgconfig /usr/local/share/aclocal /usr/local/share/locale /usr/local/share/zsh /usr/local/share/zsh/site-functions /usr/local/var/homebrew/locks
+if [[ -z "$NOROOT" ]]; then
+  sudo chown -R $(whoami) /usr/local/Cellar
+  sudo chown -R $(whoami) /usr/local/Homebrew
+  sudo chown -R $(whoami) /usr/local/var/homebrew
+  sudo chown -R $(whoami) /usr/local/etc/bash_completion.d /usr/local/include /usr/local/lib/pkgconfig /usr/local/share/aclocal /usr/local/share/locale /usr/local/share/zsh /usr/local/share/zsh/site-functions /usr/local/var/homebrew/locks
+else
+  echo "No root: Skipping homebrew folders permission check"
```

Comment:
Should we add a caution that the build may fail if another user takes ownership of these files?

Reply:
Good point, pls take a look at new commit. I think I improved the error reporting. If you think it's unnecessary change I'll revert and just extend the origin error message.

Reply:
Thanks. This looks good now. Not related to this PR, but looks like CI build is failing for Linux and Android, PR can be merged once that is fixed. Looking into it now.

Reply:
Can you now try to rebase the PR branch with main, to validate if the CI build is successful?

Reply:
Done, let's see how CI will go :)

Reply:
Thanks. Should be good to merge. CodeQL scan is failing, #967 will address this.

Reply:
Should be good to merge.  The CodeQL failure can be ignored for now, we have issue #967 for that.

Reply:
cool, can you pls merge it (I don't have write access, just a visitor here :)) or we're waiting for more reviewers?